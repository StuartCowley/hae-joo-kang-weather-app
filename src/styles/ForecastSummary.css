.forecast-summary {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  padding: 30px;
  border-top: 1px solid gray;
  border-bottom: 1px solid gray;
}

.forecast-summary__icon {
  padding-top: 15px;
  font-size: 7vh;
}

.forecast-summary__button {
  margin-top: 15px;
}