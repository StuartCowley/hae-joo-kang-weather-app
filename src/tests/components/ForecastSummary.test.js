import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    forecast: {
      date: 1525046400000,
      description: "Stub description",
      icon: "800",
      temperature: {
        min: 4,
        max: 11,
      },
    },
    onSelect: jest.fn(),
  };

  it("renders correctly", () => {
    const rendered = renderer.create(
      <ForecastSummary
        forecast={validProps.forecast}
        onSelect={validProps.onSelect}
      />
    );

    expect(rendered).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    render(
      <ForecastSummary
        forecast={validProps.forecast}
        onSelect={validProps.onSelect}
      />
    );

    expect(screen.getByText("Mon Apr 30")).toHaveAttribute(
      "class",
      "forecast-summary__date"
    );
    expect(screen.getByText("Stub description")).toHaveAttribute(
      "class",
      "forecast-summary__description"
    );
    expect(screen.getByTestId("forecast-icon")).toHaveClass(
      "forecast-summary__icon"
    );
    expect(screen.getByText("11Â°C")).toHaveAttribute(
      "class",
      "forecast-summary__temperature"
    );
  });

  it("clicking forecast calls correct function", () => {
    render(
      <ForecastSummary
        forecast={validProps.forecast}
        onSelect={validProps.onSelect}
      />
    );

    fireEvent.click(screen.getByRole("button"));
    expect(validProps.onSelect).toHaveBeenCalled();
    expect(validProps.onSelect).toHaveBeenCalledTimes(1);
    expect(validProps.onSelect).toHaveBeenCalledWith(validProps.forecast.date);
  });
});
