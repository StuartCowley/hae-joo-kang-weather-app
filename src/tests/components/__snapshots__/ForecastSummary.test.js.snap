// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<div
  className="forecast-summary"
  data-testid="forecast-summary"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="forecast-summary__date"
  >
    Mon Apr 30
  </div>
  <div
    className="forecast-summary__icon"
    data-testid="forecast-icon"
  >
    <i
      className="wi wi-day-sunny"
      name="owm"
    />
  </div>
  <div
    className="forecast-summary__temperature"
  >
    11
    Â°C
  </div>
  <div
    className="forecast-summary__description"
  >
    Stub description
  </div>
</div>
`;
